TrackletAlgorithm/MatchEngine.cc

```diff
@@ -160,27 +162,34 @@ void MatchEngine(const BXType bx, BXType& bx_o,
 			}
 
 			// Read stub memory and extract data fields
-			auto const stubadd   = zbin.concat(istubtmp);
-			auto const stubdata  = inputStubData.read_mem(bx,stubadd);
-			auto const stubindex = stubdata.getIndex();
-			auto const stubfinez = stubdata.getFineZ();
-			auto const stubbend  = stubdata.getBend();
+			auto const stubadd     = zbin.concat(istubtmp);
+			auto const stubdata    = inputStubData.read_mem(bx,stubadd);
+			auto const stubindex   = stubdata.getIndex();
+			auto const stubfinez   = stubdata.getFineZ();
+			auto const stubfinephi = stubdata.getFinePhi();
+			auto const stubbend    = stubdata.getBend();
 
 			// Calculate fine z position
 			ap_int<VMProjectionBase<PROJECTIONTYPE>::kVMProjFineZSize+1> projfinezadj = projfinez;
 			if (second) projfinezadj = projfinezadj - kZAdjustment;
 			ap_int<VMProjectionBase<PROJECTIONTYPE>::kVMProjFineZSize+1> idz          = stubfinez - projfinezadj;
+			ap_int<VMProjectionBase<PROJECTIONTYPE>::kVMProjFinePhiSize+1> idphi      = projfinephi - stubfinephi;
 
 			// Check if stub z position consistent
 			bool pass = (isPSseed) ? (idz >= ME::StubZPositionBarrelConsistency::kPSMin && idz <= ME::StubZPositionBarrelConsistency::kPSMax)
 								   : (idz >= ME::StubZPositionBarrelConsistency::k2SMin && idz <= ME::StubZPositionBarrelConsistency::k2SMax);
 
+			// Check is stub phi positions are consistent
+			bool passphi = ((idphi<ME::StubPhiPositionBarrelConsistency::kMin)&&(idphi>-ME::StubPhiPositionBarrelConsistency::kMin)) || 
```

Comment:
@aperloff Is this expression right? It just reads a little weird because the absolute value of `idphi` is required to _not_ be in [kMin, kMax].

Reply:
@aehart I think you added this expression just a little while ago:
https://github.com/cms-L1TK/firmware-hls/commit/231e50c8b1f87ecb01379799ec50e9a8a8cc2e9c
I hadn't seen this line before you made the change and I assumed you knew what you were doing.

Reply:
This was the result of a squash merge that I needed to do after some rebase chaos. So we lost authorship information for the commits that were squashed. But it sounds like you didn't add this line?

@aryd Did you add this line, or do you know who did?

Reply:
Correct, I did **not** add this line.